src/tests/Cookbook/Validate.cs

```diff
@@ -74,6 +75,21 @@ public void Test_CookbookRecipes_IDsAreValid()
             }
         }
 
+        [Test]
+        public void Test_CookbookRecipes_TitlesInsideACategoryAreUnique()
+        {
+            foreach (var cat in ScottPlot.Cookbook.Locate.GetCategorizedRecipes())
+            {
+                List<string> seen = new List<string>();
```

Comment:
This is admittedly nitpicky considering this likely has a meaningless performance impact as the categories likely contain less than a dozen demos. However, as the order is unimportant to this problem, and the only operations applied to the collection are adding elements and checking for membership, a `HashSet` would be a better choice.

`HashSet` uses black magic (hash functions) to give constant time membership checking. A `List` gives linear time. If you're unfamiliar a set basically can be thought of as a dictionary with only keys, no values.

Reply:
For what it's worth, if you need ordered elements you can use a `SortedSet` which stores elements in _sorted_ order (but you can provide whichever `IComparer` you like). This is (presumably) implemented as a Binary Search Tree, as it gives O(log n) search. `SortedSet` still does not allow duplicates, you'd have to create your own search tree to do that.

Reply:
Hi @Benny121221, thanks for the suggestion! I made this change in e2275a8

I don't usually try to optimize cookbook code (or most code outside hot loops), but this conversation is an excellent reminder that a strong understanding of the `System.Collections.Generic` types can be very useful! Thanks for bringing it up

https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic?view=net-5.0